src/shared/dropdown-list/types.ts

```diff
@@ -6,6 +6,8 @@ export interface DropdownDisplayExtractorProps<T, V> {
     valueExtractor?: (item: T) => V | undefined;
     /** Function to derive options display value from an item */
     listExtractor?: (item: T) => string | undefined;
+    /** Function to render custom component */
+    renderListItem?: (item: T) => JSX.Element | undefined;
```

Comment:
might want to rename `DropdownDisplayExtractorProps` to `DropdownDisplayProps` because this interface is no longer strictly for extractors